### PR TITLE
Bluetooth: audio: ascs: Fix handling zero Number_of_ASEs parameter

### PR DESCRIPTION
**Pull request by MariuszSkamra**
As per ASCS_v1.0
"A client-initiated ASE Control operation shall be defined as an
invalid length operation if the Number_of_ASEs parameter value is
less than 1".

Fixes: ASCS/SR/SPE/BI-{02,03,04,05,06}-C
Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

### CODE REVIEW

**Comment by MariuszSkamra** [comment]
@szymon-czapracki FYI

File: subsys/bluetooth/audio/ascs.c [hunk 1]
```diff
@@ -1312,7 +1312,10 @@ static ssize_t ascs_config(struct bt_ascs *ascs, struct net_buf_simple *buf)
 
 	BT_DBG("num_ases %u", req->num_ases);
 
-	if (buf->len < req->num_ases * sizeof(*cfg)) {
+	if (req->num_ases < 1) {
+		BT_ERR("Number_of_ASEs parameter value is less than 1");
```
Comment (jhedberg) [inline subsys/bluetooth/audio/ascs.c:1316]: I realize that the issue already existed in this file, but BT_ERR() should really be reserved for local errors only. Whenever a remote peer sends something invalid that we cleanly handle then BT_WARN() is the right choice.

**Comment by jhedberg** [comment]
@MariuszSkamra do you want to update the `else if` branches as well (perhaps a separate commit in this PR)? You now have a mix of BT_WARN/BT_ERR which is not ideal..

**Comment by MariuszSkamra** [comment]
> @MariuszSkamra do you want to update the `else if` branches as well (perhaps a separate commit in this PR)? You now have a mix of BT_WARN/BT_ERR which is not ideal..

I've pushed it as separate commit

**Review by jhedberg — APPROVED** [review]

**Review by sjanc — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**